FrameworkSystem/Service/SystemAdministratorHandler.py

```diff
@@ -54,7 +55,12 @@ def initializeHandler( cls, serviceInfo ):
       global gMonitoringReporter
       gMonitoringReporter = MonitoringReporter( monitoringType = "ComponentMonitoring" )
       gThreadScheduler.addPeriodicTask( 120, cls.__storeProfiling )
-      
+    
+    automaticCleanup = cls.srv_getCSOption( 'AutomaticCleanup', -1 )
```

Comment:
Option name is not informative, cleanup can be anything. The option has a meaning of the number of software versions to keep. So, the name can be KeepSoftwareVersions, for example. 
The default value should be 0 ? Otherwise the cleanup task will be created with -1 versions to keep.